ct_optcon/include/ct/optcon/solver/lqp/HPIPMInterface-impl.hpp

```diff
@@ -22,12 +22,26 @@ HPIPMInterface<STATE_DIM, CONTROL_DIM>::HPIPMInterface() : N_(-1), settings_(NLO
 template <int STATE_DIM, int CONTROL_DIM>
 HPIPMInterface<STATE_DIM, CONTROL_DIM>::~HPIPMInterface()
 {
-    // todo is there memory that needs to be freed?
+    if (dim_mem_) {
```

Comment:
it seems that it would make sense to collect these (dublicate) lines of code in a private function, you could call it something like freeHpipmMemory() or similar

Reply:
Done, I implemented the `freeHpipmMemory()` function.